frontend/pkg/frontend/context.go

```diff
@@ -123,17 +123,33 @@ func SystemDataFromContext(ctx context.Context) (*arm.SystemData, error) {
 	return systemData, nil
 }
 
-func ContextWithSubscriptionState(ctx context.Context, subscriptionState arm.RegistrationState) context.Context {
-	return context.WithValue(ctx, contextKeySubscriptionState, subscriptionState)
+func ContextWithSubscription(ctx context.Context, subscription arm.Subscription) context.Context {
+	return context.WithValue(ctx, contextKeySubscription, subscription)
 }
 
 func SubscriptionStateFromContext(ctx context.Context) (arm.RegistrationState, error) {
```

Comment:
Sorry, I should have been clearer.  Implicit in my suggestion was to also replace this and `TenantIDFromContext` with `SubscriptionFromContext`.  Unless you disagree?

Reply:
No qualms with adding `SubscriptionFromContext` - I can change it

Reply:
I felt like keeping `TenantIDFromContext` could make sense to centralize the error handling around the optional fields, what do you think?

Reply:
SGTM  :+1: